### ISSUE TITLE
cannot add pgp key with --add-pgp -i

### ISSUE THREAD

**Issue by davidovich** [mozilla/sops#550]
Trying to add in-place a pgp key to an existing encrypted file fails:

```
$ sops --version
sops 3.4.0 (latest)
```
```
$ sops -i --verbose --add-pgp E35533BFB792534A92A86538BFFF066934627290 secrets-az.enc.json 
[GCPKMS]         INFO[0000] Decryption failed                             resourceID=projects/..[redacted]/...[redacted]/sops-key
[PGP]    INFO[0000] Decryption succeeded                          fingerprint=AFD66F2CE2CD25473FED0046806D21D0061678DD
[SOPS]   INFO[0000] Data key recovered successfully              
[SOPS]   DEBU[0000] Decrypting tree                              
[CMD]    INFO[0001] File written successfully    
```
The editor is summoned and saving and exiting only modifies the lastmodifed field.

I would expect that:
1) the Editor is not invoked (form the `-i` flag)
2) my pgp key is added to the pgp keys array.

Both expectations fails.

**Comment by davidovich** [comment]
Yes, I have observed that adding `-r` works. But what if I do not want to rotate the encryption key ?

`-i` should suppress the editor no ?

**Comment by autrilla** [comment]
No, -i means in place, meaning the output file is the input file.


You could use the sops updatekeys command to add or remove keys without
changing the master key.

**Comment by davidovich** [comment]
@autrilla AFAICT `updatekeys` is not documented at the moment, and the help is a bit shallow on how I could use it to update (or in my case add) new keys. It mentions using a config file. Assuming it is the same format as the `.sops.yaml` file, could a key be added by using this content ?:

```
creation_rules:
      pgp: 'FBC7B9E2A4F9289AC0C1D4843D16CEE4A27381B4'
```

A am no security expert, but I still think that `--add-pgp` should not be coupled to `-r` as it is [right now in the code](https://github.com/mozilla/sops/blob/14a22d7a7060a0fba06c5dd716898d47728f0fce/cmd/sops/main.go#L582-L590).

**Comment by autrilla** [comment]
There is only one SOPS config file, and that is indeed `.sops.yaml`.

> Assuming it is the same format as the `.sops.yaml` file, could a key be added by using this content ?:
> 
> ```
> creation_rules:
>       pgp: 'FBC7B9E2A4F9289AC0C1D4843D16CEE4A27381B4'
> ```

That probably doesn't do what you want. That will make the file encrypted only with `FBC7B9E2A4F9289AC0C1D4843D16CEE4A27381B4`, and will remove all your other keys. You'll need to list all keys present in your file.

> A am no security expert, but I still think that --add-pgp should not be coupled to -r as it is right now in the code.

Personally I feel that `--add-pgp` and friends should just be removed in favor of updatekeys, because your key configuration should be in a config file, so at this point I'm not inclined to adding `--add-pgp` and such to other commands. I'm open to be convinced otherwise, though.

**Comment by davidovich** [comment]
Ok, I was able to streamline this use case with your help:

First, update the `.sops.yaml` file with comma separated pgp keys, and (in my case) gcp_kms resource_id:

```
creation_rules:
  - pgp: '[First_fingerprint],[second_fingerprint],[third_fingerprint]'
    gcp_kms: projects/...[redacted].../sops-key
```

then:

```
sops updatekeys secrets-az.enc.json 
```

And this works. If you are interested, I will submit a documentation PR for this feature.

**Comment by autrilla** [comment]
Documentation is always good :), feel free to submit a PR for this. I was working on new docs for the project (https://github.com/mozilla/sops/tree/docs/docs) but then life got in the way. Maybe I'll finish it some day...

**Comment by davidovich** [comment]
Closed by documentation update: #552
